src/core/Crust/hooks.tsx

```diff
@@ -0,0 +1,110 @@
+import { ReactNode, RefObject, useCallback, useMemo, useRef, useState } from "react";
+
+import { BlockProps } from "./Infobox";
+import Plugin, { ExternalPluginProps } from "./Plugins/Plugin";
+import { PluginModalInfo } from "./Plugins/Plugin/ModalContainer";
+import { PluginPopupInfo } from "./Plugins/Plugin/PopupContainer";
+import { MapRef } from "./types";
+import { WidgetProps } from "./Widgets";
+
+export default function useHook({
+  mapRef,
+  pluginBaseUrl,
+  pluginProperty,
+}: { mapRef?: RefObject<MapRef> } & ExternalPluginProps) {
+  const [shownPluginModalInfo, onPluginModalShow] = useState<PluginModalInfo>();
+  const pluginModalContainerRef = useRef<HTMLDivElement>();
+
+  const [shownPluginPopupInfo, onPluginPopupShow] = useState<PluginPopupInfo>();
+  const pluginPopupContainerRef = useRef<HTMLDivElement>();
```

Comment:
These lines are copied from [Visualizer/hooks.ts](https://github.com/reearth/reearth-web/blob/ff47c54a53cdf165425a522625f255c9247abd26/src/components/molecules/Visualizer/hooks.ts#L270-L274). Because this is only used by plugin system.